### PR TITLE
[11.0] [MIG] sale_procurement_group_by_line

### PR DESCRIPTION
**Pull request by murtuzasaleh**
Sale Procurement Group by line
==============================

- This module was written to extend the functionality of procurement groups
created from a Sale order.

- On itself, this module does nothing it is a requirement for modules which
needs to create procurement group per sale order line basis.

Migration of module Sale Procurement Group by line v10 to v11.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@jbeficent is this intended for the same as https://github.com/OCA/purchase-workflow/pull/563?

File: sale_procurement_group_by_line/__manifest__.py
Comment (JayVora-SerpentCS) [inline sale_procurement_group_by_line/__manifest__.py]: Full name plz

File: sale_procurement_group_by_line/__manifest__.py
Comment (JayVora-SerpentCS) [inline sale_procurement_group_by_line/__manifest__.py]: pep-8

**Review by JayVora-SerpentCS — CHANGES_REQUESTED** [review]

**Comment by murtuzasaleh** [comment]
@pedrobaeza @mreficent Could you please check it.

**Review by nikul-serpentcs — COMMENTED** [review]
Code Review LGTM

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM

**Comment by nikul-serpentcs** [comment]
@pedrobaeza @JayVora-SerpentCS Could you please review and move forward.

**Comment by pedrobaeza** [comment]
I'm afraid I don't know this module, so others should review it.

File: sale_procurement_group_by_line/README.rst
Comment (HviorForgeFlow) [inline sale_procurement_group_by_line/README.rst:16]: Improve usage section (add an usage example, where you canbenefit from this module avoid using technical language), add some pictures, would be nice to move readme to [new format](https://github.com/OCA/maintainer-tools/tree/master/template/module/readme)

File: sale_procurement_group_by_line/__init__.py
Comment (HviorForgeFlow) [inline sale_procurement_group_by_line/__init__.py]: remove copyright from __init__ files.

File: sale_procurement_group_by_line/model/__init__.py
Comment (HviorForgeFlow) [inline sale_procurement_group_by_line/model/__init__.py]: Same here

File: sale_automatic_workflow_payment_mode/tests/test_automatic_workflow_payment_mode.py
Comment (HviorForgeFlow) [inline sale_automatic_workflow_payment_mode/tests/test_automatic_workflow_payment_mode.py:52]: That file is not related with this module, please remove

File: sale_procurement_group_by_line/tests/__init__.py
Comment (HviorForgeFlow) [inline sale_procurement_group_by_line/tests/__init__.py]: Same here

File: sale_procurement_group_by_line/model/sale.py
Comment (HviorForgeFlow) [inline sale_procurement_group_by_line/model/sale.py:108]: As an UserError is caught, maybe it's interesting to append  which products have raised it.

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

**Comment by JayVora-SerpentCS** [comment]
@murtuzasaleh check.

File: sale_procurement_group_by_line/model/sale.py
Comment (mreficent) [inline sale_procurement_group_by_line/model/sale.py]: add
```python
res = super(SaleOrderLine, self)._action_launch_procurement_rule()
```
and then end the method with a `return res` instead of `return True`

File: sale_procurement_group_by_line/model/sale.py
Comment (MiquelRForgeFlow) [inline sale_procurement_group_by_line/model/sale.py]: Why did you delete this super?

**Review by MiquelRForgeFlow — COMMENTED** [review]
@murtuzasaleh will you finish this PR?

File: sale_procurement_group_by_line/model/sale.py
Comment (murtuzasaleh) [reply sale_procurement_group_by_line/model/sale.py]: @mreficent Because In odoo v11 _action_procurement_create method is removed and added _action_launch_procurement_rule method [https://github.com/odoo/odoo/commit/c914df5a369eb49e4e72a698c0811200b5825f22#diff-bf6a9af0e4d7537746d73deb353f3f8aL210](https://github.com/odoo/odoo/commit/c914df5a369eb49e4e72a698c0811200b5825f22#diff-bf6a9af0e4d7537746d73deb353f3f8aL210) so I have overwrite this method and added code.

File: sale_procurement_group_by_line/model/sale.py
Comment (AaronHForgeFlow) [reply sale_procurement_group_by_line/model/sale.py:108]: If super cannot be called then you need to add a comment telling clearly that you are overwriting the original method. Also include it in the known issues section.

**Review by AaronHForgeFlow — COMMENTED** [review]
Code seems ok to me, it is not calling super but conserving the original behavior, plus adding the grouping by key feature.

File: sale_procurement_group_by_line/model/sale.py
Comment (MiquelRForgeFlow) [reply sale_procurement_group_by_line/model/sale.py]: Yeah, but why you remove the super? You can do super with the _action_launch_procurement_rule method.

Comment (nikul-serpentcs) [reply sale_procurement_group_by_line/model/sale.py]: @mreficent [Here](https://github.com/odoo/odoo/commit/c914df5a369eb49e4e72a698c0811200b5825f22#diff-bf6a9af0e4d7537746d73deb353f3f8aR235) super return `True` , So How can modify the code with super?.

That's way @murtuzasaleh overwrite code and modify that.

Here Super is not need. @aheficent already https://github.com/OCA/sale-workflow/pull/673#discussion_r233780204

Comment (MiquelRForgeFlow) [reply sale_procurement_group_by_line/model/sale.py]:
```python
return super(self, SaleOrderLine)._action_launch_procurement_rule()
```

Comment (AaronHForgeFlow) [reply sale_procurement_group_by_line/model/sale.py]: Hi @murtuzasaleh @nikul-serpentcs Did you try to call super at the end? what does the system do in that case? Are other procurement groups created or the ones you create updated? Maybe they are not. I am not sure looking at the code.

Comment (nikul-serpentcs) [reply sale_procurement_group_by_line/model/sale.py]: >  Did you try to call super at the end?

@aheficent Yes, But it's method Call again.

Comment (AaronHForgeFlow) [reply sale_procurement_group_by_line/model/sale.py]: But does that affect the result?

**Comment by LoisRForgeFlow** [comment]
@murtuzasaleh Can you rebase? Error in CI was due to another module, but it's fixed now.

**Comment by AaronHForgeFlow** [comment]
BTW sale_source_by_line will be deprecated. This, this module is only a dependency for this one #668 Do you agree @murtuzasaleh?

**Comment by MiquelRForgeFlow** [comment]
Superseeded by https://github.com/OCA/sale-workflow/pull/758.

**Pull request closed without merging**
